### PR TITLE
[INLONG-3868][Sort] Support data from MySQL binlog to Kafka

### LINKED ISSUE #3868: [Feature][Sort] Support data from mysql binlog sync to kafka
### Description

Support data from mysql binlog sync to kafka

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yunqingmoswu**
### Title Name: [INLONG-3868][Sort]Support data from mysql binlog sync to kafka

Fixes #3868 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-single-tenant/pom.xml [hunk 1]
```diff
@@ -166,6 +166,18 @@
             <artifactId>hadoop-minicluster</artifactId>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>com.ververica</groupId>
+            <artifactId>flink-connector-mysql-cdc</artifactId>
+            <version>2.0.2</version>
```
Comment (healchow) [inline inlong-sort/sort-single-tenant/pom.xml:173]: Please move the dependent version number to the pom file in the root directory.

Comment (yunqingmoswu) [reply inlong-sort/sort-single-tenant/pom.xml]: done

File: inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/parser/impl/FlinkSqlParser.java [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+            }
+            if (sb.length() > 0) {
+                sb.delete(sb.lastIndexOf(","), sb.length());
+            }
+            sb.append("\n)");
+        }
+        return sb.toString();
+    }
+
+    public String parseTransformNodeFields(List<FieldInfo> fields) {
```
Comment (healchow) [inline inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/parser/impl/FlinkSqlParser.java:322]: This method was `public`, does it necessary?

File: inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/parser/result/FlinkSqlParseResult.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        this.loadSqls = Preconditions.checkNotNull(loadSqls, "loadSqls is null");
+        Preconditions.checkState(!loadSqls.isEmpty(), "loadSqls is empty");
+    }
+
+    /**
+     * Execute the parse result
+     *
+     * @throws Exception The exception may throws when executing
+     */
+    @Override
```
Comment (healchow) [inline inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/parser/result/FlinkSqlParseResult.java:63]: In general, the override method does not need to add java docs, but only needs to be added to the method in the interface.

File: pom.xml [hunk 4]
```diff
@@ -193,6 +193,7 @@
         <kafka.version>2.4.1</kafka.version>
         <iceberg.flink.version>0.12.1</iceberg.flink.version>
         <flink.version>1.13.5</flink.version>
+        <mysql.cdc.version>2.0.2</mysql.cdc.version>
```
Comment (healchow) [inline pom.xml:196]: Maybe call it with `flink.connector.mysql.cdc.version` is more clear?

File: inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/parser/impl/FlinkSqlParser.java (hunk 2 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/parser/impl/FlinkSqlParser.java:322]: It is not necessary, I'll deal with it later.

File: inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/parser/result/FlinkSqlParseResult.java (hunk 3 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/parser/result/FlinkSqlParseResult.java:63]: done

File: inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/parser/impl/FlinkSqlParser.java (hunk 2 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/parser/impl/FlinkSqlParser.java]: done

**Review by healchow — APPROVED** [review]
LGTM

**Review by EMsnap — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Pull request merged**